src/watertap_contrib/seto/multiperiod/VAGMD_batch_flowsheet.py

```diff
@@ -0,0 +1,255 @@
+from pyomo.environ import (
+    ConcreteModel,
+    Var,
+    Objective,
+    Param,
+    Expression,
+    Constraint,
+    Block,
+    log10,
+    TransformationFactory,
+    assert_optimal_termination,
+    value,
+    units as pyunits,
+)
+
+# IDAES imports
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.core.util.scaling as iscale
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers.get_solver import get_solver
+import idaes.logger as idaeslog
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+    badly_scaled_var_generator,
+)
+
+# WaterTAP imports
+from watertap.property_models.seawater_prop_pack import SeawaterParameterBlock
+from watertap_contrib.seto.unit_models.surrogate import VAGMDSurrogate
+
+
+def build_vagmd_flowsheet(
+    m=None,
+    feed_flow_rate=600,
+    evap_inlet_temp=80,
+    cond_inlet_temp=25,
+    feed_temp=25,
+    feed_salinity=35,
+    initial_batch_volume=50,
+    module_type="AS7C1.5L",
+    high_brine_salinity=False,
+    cooling_system_type="closed",
+):
+    """
+    This function builds a unit model for a certain time period
+
+    Returns:
+        object: A Pyomo concrete optimization model and flowsheet
+    """
+    if m is None:
+        m = ConcreteModel()
+
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.properties = SeawaterParameterBlock()
+    m.fs.vagmd = VAGMDSurrogate(
+        property_package=m.fs.properties,
+        module_type=module_type,
+        high_brine_salinity=high_brine_salinity,
+        cooling_system_type=cooling_system_type,
+    )
+
+    # Fix the model inputs
+    m.fs.vagmd.feed_props[0].flow_vol_phase["Liq"].fix(
+        pyunits.convert(
+            feed_flow_rate * pyunits.L / pyunits.h, to_units=pyunits.m**3 / pyunits.s
+        )
+    )
+    m.fs.vagmd.evaporator_in_props[0].temperature.fix(evap_inlet_temp + 273.15)
+
+    if cooling_system_type == "closed":  # TODO: update closed cooling
+        m.fs.vagmd.condenser_in_props[0].temperature.fix(cond_inlet_temp + 273.15)
+    else:  # "open"
+        m.fs.vagmd.condenser_in_props[0].temperature.fix(cond_inlet_temp + 273.15)
+
+    """
+    Variables inherited from previous period
+    """
+    m.fs.pre_feed_salinity = Var(
+        initialize=35,
+        bounds=(0, 292),
+        units=pyunits.g / pyunits.L,
+        doc="Feed salinity from previous time step",
+    )
+    m.fs.pre_feed_temperature = Var(
+        initialize=298.15,
+        bounds=(273.15, 373.15),
+        units=pyunits.K,
+        doc="Feed temperature from previous time step",
+    )
+    m.fs.pre_evap_out_temp = Var(
+        initialize=308.15,
+        bounds=(273.15, 373.15),
+        units=pyunits.K,
+        doc="Feed temperature from previous time step",
+    )
+    m.fs.pre_permeate_flow_rate = Var(
+        initialize=1e-5,
+        units=pyunits.m**3 / pyunits.s,
+        doc="Permeate flow rate from previous time step",
+    )
```

Comment:
I wonder if these can't be defined through a stateblock or two since they are properties.

Reply:
I had the same thought, but I don't know enough about the multiperiod framework to know if this is easier or harder.

Reply:
We can ignore this for now since this is a special case and might be more straightforward than making a bunch of stateblocks.